diff --git a/source/build/src/sdlayer.cpp b/source/build/src/sdlayer.cpp
--- a/source/build/src/sdlayer.cpp
+++ b/source/build/src/sdlayer.cpp
@@ -1631,7 +1631,9 @@ void sdlayer_setvideomode_opengl(void)
 
     glShadeModel(GL_SMOOTH);  // GL_FLAT
     glClearColor(0, 0, 0, 1.0);  // Black Background
+#ifndef __EMSCRIPTEN__
     glHint(GL_PERSPECTIVE_CORRECTION_HINT, GL_NICEST);  // Use FASTEST for ortho!
+#endif
 //    glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
     glViewport(0,0,xres,yres);
 
